### PR TITLE
Fix textual message stripping new line

### PR DESCRIPTION
**Pull request by renancleyson-dev**
Signed-off-by: Renan <renancleyson.f@gmail.com>

Fixes vector-im/element-web#15320

# Proposed changes
Element doesn't render HTML on plain text but still adds a `formatted_body`(markdown) to the message's content when there's escaped markdown to remove the backslashes, so we only add the `markdown-body` CSS class(which is currently added when there's `formatted_body`, even if it's only plain text) when it's a `formatted body` with HTML as the bug is caused by the CSS rule `white-space: normal` added by this class.

### Before
WhatsApp Image 2021-11-30 at 20 38 42

### After
WhatsApp Image 2021-11-30 at 20 37 04

# Steps to reproduce
1. Send a multiline plain text message with escaped markdown
2. Newlines are converted to white-spaces

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix textual message stripping new line ([\#7239](https://github.com/matrix-org/matrix-react-sdk/pull/7239)). Fixes vector-im/element-web#15320. Contributed by @renancleyson-dev.

Preview: https://61a7bf9bf8f071611e704c77--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by andybalaam** [comment]
So you've got some failing tests to look at.

What would be totally awesome is if you wrote a new test that failed before your change, and passed after it.

File: src/HtmlUtils.tsx [hunk 1]
```diff
@@ -449,20 +450,23 @@ export function bodyToHtml(content: IContent, highlights: string[], opts: IOpts
...
             isDisplayedWithHtml = true;
+
             safeBody = sanitizeHtml(formattedBody, sanitizeParams);
+            const phtml = cheerio.load(safeBody, {
+                // @ts-ignore: The `_useHtmlParser2` internal option is the
+                // simplest way to both parse and render using `htmlparser2`.
+                _useHtmlParser2: true,
+                decodeEntities: false,
+            });
+            const isPlainText = phtml.html() === phtml.root().text();
```
Comment (renancleyson-dev) [inline src/HtmlUtils.tsx:466]: I didn't know that there are some situations where the content body from a message can be plain text while the formatted body doesn't(e.g. mentions). Trying to rely only on the formatted body now.

**Review by andybalaam — APPROVED** [review]
Brilliant, thanks!

**Comment by ShadowJonathan** [comment]
:tada:

**Pull request merged**